### ISSUE TITLE
Testing infrastructure

### ISSUE THREAD

**Issue by antoine-levitt** [JuliaMolSim/DFTK.jl#491]
We should probably at some point overhaul our test suite. Eg it'd be nice to decouple the features from the systems. There could be a list of predefined pwbasis together with some characteristics (metal, insulator, small, large, symmetries, floating point type, etc), the tests would define test_feature(basis) and there'd be a runner in feature.jl that runs test_feature on the systems that are sensible for this particular test.

(low-priority as the current system good enough)

**Comment by mfherbst** [comment]
Yes I thought about that. I think a good way forward could be to have some defined input files for the systems (e.g. in QE format or in "our own") and use that for the system definition. That allows to generate test data / compare against a standard code.

But I agree, I think our test system is pretty solid and it's easy to put a lot of effort into this for only getting marginally better. Doing this generically such that we do it once and don't need to bother about it for the time to come is not so easy.
